File: src/test/java/ru/spasitel/factorioautoplanner/simple/MainTest.java
package ru.spasitel.factorioautoplanner.simple;

import org.junit.jupiter.api.Test;
import ru.spasitel.factorioautoplanner.simple.building.Beacon;
import ru.spasitel.factorioautoplanner.simple.building.Smelter;

import static org.junit.jupiter.api.Assertions.assertFalse;
import static org.junit.jupiter.api.Assertions.assertTrue;

class MainTest {

    @Test
    public void intersectTest() {
        assertTrue(Main.isInteract(
                new Smelter(1, 0),
                new Smelter(0, 1)));
        assertFalse(Main.isInteract(
                new Smelter(0, 1),
                new Smelter(0, 4)
        ));
        for (int i = 1; i < 6; i++) {
            for (int k = 1; k < 6; k++) {
                assertTrue(Main.isInteract(
                        new Smelter(3, 3),
                        new Smelter(i, k)));
                assertTrue(Main.isInteract(
                        new Beacon(3, 3),
                        new Smelter(i, k)));
                assertTrue(Main.isInteract(
                        new Smelter(3, 3),
                        new Beacon(i, k)));
                assertTrue(Main.isInteract(
                        new Beacon(3, 3),
                        new Beacon(i, k)));
            }
        }

        assertTrue(Main.isInteract(
                new Smelter(3, 3),
                new Smelter(6, 1)
        ));


    }
}

File: src/test/kotlin/ru/spasitel/factorioautoplanner/simple/MainTest.kt
package ru.spasitel.factorioautoplanner.simple

import org.junit.jupiter.api.Assertions
import org.junit.jupiter.api.Test
import ru.spasitel.factorioautoplanner.simple.Main.isInteract
import ru.spasitel.factorioautoplanner.simple.building.Beacon
import ru.spasitel.factorioautoplanner.simple.building.Smelter

internal class MainTest {
    @Test
    fun intersectTest() {
        Assertions.assertTrue(
            isInteract(
                Smelter(1, 0),
                Smelter(0, 1)
            )
        )
        Assertions.assertFalse(
            isInteract(
                Smelter(0, 1),
                Smelter(0, 4)
            )
        )
        for (i in 1..5) {
            for (k in 1..5) {
                Assertions.assertTrue(
                    isInteract(
                        Smelter(3, 3),
                        Smelter(i, k)
                    )
                )
                Assertions.assertTrue(
                    isInteract(
                        Beacon(3, 3),
                        Smelter(i, k)
                    )
                )
                Assertions.assertTrue(
                    isInteract(
                        Smelter(3, 3),
                        Beacon(i, k)
                    )
                )
                Assertions.assertTrue(
                    isInteract(
                        Beacon(3, 3),
                        Beacon(i, k)
                    )
                )
            }
        }
        Assertions.assertTrue(
            isInteract(
                Smelter(3, 3),
                Smelter(6, 1)
            )
        )
    }
}